From: "Stefan Lippers-Hollmann" <s.L-H@gmx.de>
To: Jesse Barnes <jbarnes@virtuousgeek.org>
Subject: Re: Patch "drm/i915: fix ILK+ infoframe support" has been added to the 3.1-stable tree
Date: Fri, 4 Nov 2011 13:57:27 +0100

Hi

On Friday 04 November 2011, Jesse Barnes wrote:
> On Thu, 3 Nov 2011 17:10:43 +0100
> "Stefan Lippers-Hollmann" <s.L-H@gmx.de> wrote:
> > On Thursday 03 November 2011, Jesse Barnes wrote:
> > > On Thu, 3 Nov 2011 15:48:51 +0100
> > > "Stefan Lippers-Hollmann" <s.L-H@gmx.de> wrote:
> > [...]
> > > > > From 64a8fc0145a1d0fdc25fc9367c2e6c621955fb3b Mon Sep 17 00:00:00 2001
> > > > > From: Jesse Barnes <jbarnes@virtuousgeek.org>
> > > > > Date: Thu, 22 Sep 2011 11:16:00 +0530
> > > > > Subject: drm/i915: fix ILK+ infoframe support
[...]
> I guess current Linus master behaves the same way?

I just checked and 3.1.0-09073-gd674806 is affected the same way, yes.

> The only other bit that looks like it could cause trouble is the
> intel_hdmi_init() change:
[...]
> If you remove the writes to VIDEO_DIP_CTL do things go back to normal
> for you?  I'm just trying to figure out which part of the patch broke
> things for you (I suppose it could be the header checksum or ENABLE_SPD
> lines as well though).

After applying this patch, removing both writes to VIDEO_DIP_CTL, 
everything works as expected again.

--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -710,7 +710,6 @@ void intel_hdmi_init(struct drm_device *
 
 	if (!HAS_PCH_SPLIT(dev)) {
 		intel_hdmi->write_infoframe = g4x_write_infoframe;
-		I915_WRITE(VIDEO_DIP_CTL, 0);
 	} else if (IS_VALLEYVIEW(dev)) {
 		intel_hdmi->write_infoframe = vlv_write_infoframe;
 		for_each_pipe(i)
@@ -728,8 +727,6 @@ void intel_hdmi_init(struct drm_device *
 			I915_WRITE(TVIDEO_DIP_CTL(i), 0);
 	} else {
 		intel_hdmi->write_infoframe = cpt_write_infoframe;
-		for_each_pipe(i)
-			I915_WRITE(TVIDEO_DIP_CTL(i), 0);
 	}
 
 	if (IS_HASWELL(dev))
